packages/commerce-sdk-react/src/query-options.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {UseQueryOptions} from '@tanstack/react-query'
+
+export const getQueryOptions = <TQueryFnData, TError>(
+    userOptions?: UseQueryOptions<TQueryFnData, TError>
+): UseQueryOptions<TQueryFnData, TError> => {
+    return {
+        // Our default options at the individual query level
+        // (to override whichever options were set at the global level (e.g. when initializing the QueryClient))
+        retry: false,
```

Comment:
As the PR mentioned that `disable retries for any errors`. Would it make more sense if we have it disabledglobally, and the users can override it on the hook level if they want the retry to be enabled? 🤔

Reply:
@adamraya @alexvuong thanks for your feedback. That made me think. And I didn't know that as a team, we have discussed retries during our tech discussion meeting. I missed that meeting.

Let me get back to you, as I digest all of these info. I'm thinking now to revert the change I made. I think I should embrace more of the fact that with the current setup, the query client config is something that is not really in our control.

Reply:
@adamraya @alexvuong  I've reverted the addition of those query options. Yeah, I should embrace the fact that react-query is now configurable from the user land. So this PR no longer tries to set any `retry` or `onError` options.